### ISSUE TITLE
Changing invoice amount will sometimes cause issues

### ISSUE THREAD

**Issue by braydonf** [ACINQ/phoenix#232]
Steps to reproduce:
1. Create an invoice for something around 5 satoshis
2. Attempt to pay the invoice with another wallet, however change the amount to something greater such as 12000 satoshis.
3. The payment will fail with the error "Recipient is not reachable, or does not have enough inbound liquidity." Note: If the invoice is created for 12000 satoshis it will go through without an error.

**Comment by dpad85** [comment]
This is related to your incoming liquidity. If your existing channels don't have the liquidity to accept a incoming amount, a new LN channel will have to be created to receive the payment. However this only happens if the payment is large enough (10k sat at least) otherwise it's not worth it (because there's an on-chain tx under the hood) and payment will just fail.

See those entries in the FAQ:
- https://phoenix.acinq.co/faq#i-just-installed-the-wallet-and-i-cannot-receive-payments
- https://phoenix.acinq.co/faq#how-much-can-i-receive-with-my-existing-channels

Knowing your incoming liquidity in advance is unfortunately not possible for now so the UX can be poor. Work is being done to fix that.

**Comment by braydonf** [comment]
Okay, thanks for the info.

I believe the wallet had enough inbound capacity without the need for a new channel. We would need to test again to determine the incoming liquidity. The issue resolved immediately without the pending confirmation time that would be associated with an on-chain transaction. So far as I know, there wasn't a new channel created.

So what worked:
1. Payee creates invoice at 12000 sats and payer sends 12000 sats
2. Payee creates invoice at 5 sats and payer sends 5 sats

What did not work:
1. Payee creates invoice at 5 sats and payer sends 12000 sats

**Comment by dpad85** [comment]
> What did not work:
> 
> Payee creates invoice at 5 sats and payer sends 12000 sats

Yep an invoice cannot be overpaid by that much. At most, the protocol (and Phoenix) will let you pay twice the requested amount of an invoice, and that's only for privacy reason

See specifications: https://github.com/lightning/bolts/blob/f6c4d7604150986894bcb46d67c5c88680740b12/04-onion-routing.md#L1103 
`if the amount paid is more than twice the amount expected: SHOULD fail the HTLC.`.

**Comment by braydonf** [comment]
Bingo! That explains it.

I think this is a UX improvement: If a payer changes the amount of the invoice greater than twice, it could display a warning stating that it is too much for the requested amount. This way it does not seem that the payment failed for another reason.

**Comment by robbiehanson** [comment]
For reference, here's what it currently looks like on iOS:

<img height="450" src="https://user-images.githubusercontent.com/304604/149012462-d61526e8-8747-474b-a90b-ba7cef03c076.png">

**Comment by braydonf** [comment]
Is that something in development or already exists, just not with the Android release?
